lib/parse_identifier_doi.rb

```diff
@@ -0,0 +1,39 @@
+require_relative 'parse_identifier'
+
+# Parse the DOI identifiers in a PublicationIdentifier
+class ParseIdentifierDOI < ParseIdentifier
+
+  URI_PREFIX = 'http://dx.doi.org/'.freeze
+
+  # Does the data validate?
+  def valid?
+    value.present? && uri.present?
+  end
+
+  # Extract a DOI value
+  # @return [String|nil]
+  def value
+    @value ||= extract_value
+  end
+
+  private
+
+    # Compose a URI
+    # @return [String|nil]
+    def compose_uri
+      return nil if value.blank? || URI_PREFIX.blank?
```

Comment:
Same here, the constant can never be blank.

Reply:
This could be a copy-paste mistake, thanks.  Specs did not expose it because it's a no-op, as you say.  This can be easily fixed.

Reply:
Removed.